src/public/modules/forms/services/submissions.client.factory.js

```diff
@@ -133,8 +135,11 @@ function SubmissionsFactory(
         version: body.version,
       })
       .then(
-        function () {
-          deferred.resolve('Submission has finished.')
+        function (response) {
+          deferred.resolve({
+            message: 'Submission has finished.',
+            id: response.submissionId,
```

Comment:
would it be better to name this key `submissionId` so it's clear exactly what ID it refers to?